qt/widgets/sliceviewer/inc/MantidQtWidgets/SliceViewer/QPeaksTableModel.h

```diff
@@ -159,25 +161,18 @@ class EXPORT_OPT_MANTIDQT_SLICEVIEWER QPeaksTableModel
   static const int COL_QLAB;
   /// Index for Q-vector in the sample column
   static const int COL_QSAMPLE;
-
   /// The number of digits past the decimal to display in the table
   int m_hklPrec;
-
-  mutable std::vector<QString> m_dataCache;
-  mutable int m_dataCachePeakIndex;
-
-  QString findColumnName(const int colIndex) const;
-  void updateDataCache(const Mantid::Geometry::IPeak &peak,
-                       const int row) const;
-
+  /// Map from column index to raw peak data
+  std::vector<std::function<QVariant(const Mantid::Geometry::IPeak &)>>
```

Comment:
Does changing this to `std::vector<QVariant(*)(const MantidGeometry::IPeak &)>` work?